### PR TITLE
Reword internal errors / enable --print-callstack-on-error for users

### PR DESCRIPTION
**Pull request by bradcray**
This changes the way we print internal errors, drawing a bigger distinction between the user's view and the developer's view.  It also reorders elements of what we print and rewords the message we print for users.

Previous to this commit, a user's internal error would look like:

     Unfortunately the Chapel compiler has encountered
     an internal error. Please file a bug report that
     includes a program reproducing the problem as well
     as this output. See http://chapel.cray.com/getinvolved.html for
     further instructions on filing bug reports.

     The error may be related to this location:
     $CHPL_HOME/modules/internal/ChapelIteratorSupport.chpl:188
    $CHPL_HOME/modules/internal/ChapelIteratorSupport.chpl:188: internal error: AST0614 chpl Version 1.11.0.21dc621

and a developer's view would look like:

     Unfortunately the Chapel compiler has encountered
     an internal error. Please file a bug report that
     includes a program reproducing the problem as well
     as this output. See http://chapel.cray.com/getinvolved.html for
     further instructions on filing bug reports.
    
     The error may be related to this location:
     $CHPL_HOME/modules/internal/ChapelIteratorSupport.chpl:188
    $CHPL_HOME/modules/internal/ChapelIteratorSupport.chpl:188: [source location guessed] internal error: assertion error [astutil.cpp:614]

With this change, the user's view is:

    $CHPL_HOME/modules/internal/DefaultRectangular.chpl:751: In iterator 'these':
    $CHPL_HOME/modules/internal/ChapelIteratorSupport.chpl:163: internal error: PRI0236 chpl Version 1.11.0.52b3f06
    
    Internal errors indicate a bug in the Chapel compiler ("It's us, not you"),
    and we're sorry for the hassle.  We would appreciate your reporting this bug -- 
    please see http://chapel.cray.com/bugs.html for instructions.  In the meantime,
    the filename + line number above may be useful in working around the issue.

And the developer's view is:

    $CHPL_HOME/modules/internal/DefaultRectangular.chpl:751: In iterator 'these':
    $CHPL_HOME/modules/internal/ChapelIteratorSupport.chpl:163: internal error: bad member primitive [primitive.cpp:236]

With this change, we also now print out a note about a source location being a guess for all error types whereas before we only did it for internal errors.  This required updating two .goods that now reflect guessed locations.  (Note: A "guessed" location is one in which no AST was passed in with the error, so the location set by SET_LINENO() is used instead).

While here, I also enabled a user to throw --print-callstack-on-error without throwing --devel, because I find one of the most useful flags and often suggest users utilize it.  Though I knew that it was a developer flag, I didn't realize that its capability was disabled for users -- I thought it was only not documented (note that I haven't changed anything in terms of its documentation).

Updated .bads for .futures that are sensitive to the internal error messages.

Updated the .good for a test that reflects the current developer error message.

TODO: Create the bugs.html web page.

### CODE REVIEW

**Comment by bradcray** [comment]
This is still undergoing testing (and futures will need to be updated at the very least), but if someone wants to review what's here, I think it's the hard part and close to being done.

**Comment by bradcray** [comment]
@vasslitvinov may want to look at the .goods that changed here as a result of printing out guessed locations.
@mppf may want to look at this since it changes his previous work.

File: test/users/vass/internal-error-message.good [hunk 1]
```diff
@@ -1,8 +1,2 @@
 internal-error-message.chpl:72: In function 'scalar_outer_product_cholesky':
- Unfortunately the Chapel compiler has encountered
- an internal error. Please file a bug report that
- includes a program reproducing the problem as well
- as this output. See http://chapel.cray.com/getinvolved.html for
- further instructions on filing bug reports.
-
```
Comment (ronawho) [inline test/users/vass/internal-error-message.good]: Is this missing the new error message?

**Comment by ronawho** [comment]
Looks good to me!

**Comment by vasslitvinov** [comment]
I am concerned about the "this location is a guess" messages in
  test/users/vass/type-tests.isSubtype.good

However this is certainly acceptable for now.

**Comment by bradcray** [comment]
Hi Vass -- I was a little surprised by these as well.  I believe they appear because you requested guessed locations to be printed out for user errors as well (which seemed reasonable to me and only changed a few .goods), but it's not clear to me why some of your compilerWarnings in this test are considered guesses and others are not.  Note that I didn't change the logic for determining whether a location was a guess since the last commit -- so it could be that that logic should've been updated as well.  I'm definitely open to suggestions.

**Comment by mppf** [comment]
Looks good to me.

**Pull request merged**